typechecker/src/type_evaluator.rs

```diff
@@ -122,8 +130,21 @@ impl<'a> TypeEvaluator<'a> {
                                 scope_id,
                             );
                             Ok(return_type)
+                        } else if let PythonType::Instance(i) = &called_type {
+                            // This executes the __call__ method of the instance
+                            let Some(PythonType::Callable(c)) = self.lookup_on_class(symbol_table, &i.class_type, "__call__") else {
+                                bail!("If you call an instance, it must have a __call__ method");
+                            };
+                            let return_type = self.get_return_type_of_callable(
+                                &c,
+                                &call.args,
+                                symbol_table,
+                                scope_id,
+                            );
+                            Ok(return_type)
                         } else if let PythonType::Class(c) = &called_type {
-                            Ok(called_type)
+                            // This instantiates the class
+                            Ok(PythonType::Instance(types::InstanceType::new(c.clone(), [].to_vec())))
```

Comment:
Was this a bug? After the constructur is called, you get an instance of the class and not the class type itself